actions/actions.py

```diff
@@ -57,48 +149,45 @@ def backup() -> None:
     })
 
 
-def restore() -> None:
+def restore():
+    """
+    Implementation of easyrsa 'restore' action
+
+    Backup restoration process can be summarized as following:
+
+        * Selected backup is scanned and verified
+        * Contents of the backup are unpacked into <cahrm_dir>/EasyRSA/pki
+        * Data that are stored in the local database are updated
+        * All units that have relation with this easyrsa unit will be notified
+          about the certificate changes.
+    """
     pki_dir = os.path.join(easyrsa_directory, 'pki')
-    backup_name: str = function_get('name')
+    backup_name = function_get('name')
```

Comment:
This input should be validated against `easyrsa-\d+-\d+-\d+_\d+-\d+-\d+\.tar\.gz`

Reply:
Hmmm..I disagree here. Yes, we use the timestamped format when creating backups, but that's only because we need a way to automatically generate names that do not overlap (within reasonable timespan). But if user wants to rename the file, or upload it to a unit under a different name, I don't see why that shouldn't be possible. We perform some (very trivial) sanity checks down the line to ensure that we are not accidentally unpacking something that's not an easyrsa backup.

Reply:
OK. What about using os.path.basename? The initial reaction about input validation was to avoid running shutil.rmtree on a different path doing to the input being `../../../dir/file`. I'm fine if a different name is allowed as long as the tarball is placed where it is expected.

Reply:
I see your point, but `rmtree` is not run with user's input.
I can see a value of some basic sanitation of users input on `delete-backup` action, since we run `so.remove()` with that output. But I wouldn't put too much effort to it. If someone wants to delete their `/bin/bash` they can just ssh into the unit, no need to trick poor easyrsa :D

Reply:
Protection against directory traversal was implemented.